### ISSUE TITLE
High CPU usage after svn commands Show Log/Blame/Diff

### ISSUE THREAD

**Issue by AlexanderSemenyak** [git-as-svn/git-as-svn#669]
High CPU usage after svn commands Show Log/Blame/Diff

![image](https://github.com/user-attachments/assets/88ea0c20-6945-4da5-ba31-3ae997c02286)

Version: 4.0

**Comment by slonopotamus** [comment]
Could you show a thread dump? For example this way: https://access.redhat.com/solutions/18178#openjdk

**Comment by slonopotamus** [comment]
Also: how big is your repo? 5GB of heap might be not enough and JVM could be doing GC aggressively. Collecting JVM memory info also can be useful.

**Comment by AlexanderSemenyak** [comment]
> Also: how big is your repo? 5GB of heap might be not enough and JVM could be doing GC aggressively. Collecting JVM memory info also can be useful.

Big heap need for create map (first convert git repo to svn => out of memory, reason - maybe large commit history?). After fill map - not need big heap

**Comment by AlexanderSemenyak** [comment]
> Could you show a thread dump? For example this way: https://access.redhat.com/solutions/18178#openjdk

![image](https://github.com/user-attachments/assets/856f8798-6497-48c9-9305-c2cc9b16627d)
[jstack.zip](https://github.com/user-attachments/files/16365063/jstack.zip)

**Comment by slonopotamus** [comment]
Yup, JVM is running out of memory and is stuck in GC loop. As a quick workaround you can attempt increasing `-Xmx`.

**Comment by AlexanderSemenyak** [comment]
> Yup, JVM is running out of memory and is stuck in GC loop. As a quick workaround you can attempt increasing `-Xmx`.

Need fix in code or this behavior is expected?
